### PR TITLE
fix(breadcrumb): update aria role to now have the value of navigation

### LINKED ISSUE #10730: [YOUR TITLE]: Brief description
### Description

Hello team, I have a question, I'm having a bit of a conflict when it comes to accessibility.
The bx-breadcrumb and bx-breadcrumb-items seem to automatically add some role attributes to them, but when I run the Accessibility checker, it complains about the bx-breadcrumb assigned role, where it is role="nav" saying that "The The `role='nav'` defined on the element is not valid per ARIA specification".
I don't know if perhaps it is conflicting with the side navigation of the UI Shell component, and I wonder if I can flag this as a "False positive"
I wonder if the automatic adding of roles is intentional or not.

### Component(s) impacted

bx-breadcrumb

### Browser

Chrome

### Carbon for IBM.com version

1.28

### Severity

Severity 1 = The design is broken in a critical way that blocks users from completing tasks or damages the brand. Affects major functionality, no workaround.

### Application/website

In the storybook itself

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

https://web-components.carbondesignsystem.com/?path=/story/components-breadcrumb--default&knob-Caption%20(caption)=Time%20stamp%20[00:00:00]&knob-Color%20scheme%20(color-scheme)=light&knob-Disabled%20(disabled)=true&knob-Expanded%20(expanded)=true&knob-Helper%20text%20(helper-text)=Optional%20helper%20text&knob-Input%20size%20(size)=md&knob-Input%20type%20(type)=text&knob-Label%20text%20(label-text)=Dropdown%20title&knob-Link%20href%20(href)=javascript:void%200&knob-Open%20(open)=true&knob-Placeholder%20text%20(placeholder)=Optional%20placeholder%20text&knob-Subtitle%20(subtitle)=hilft%20mir%20bitte!%20olijasdfpoawjdpojawspo;djapowdjpoawjdpo%20apwsodjaspodjpoasjdopasjdopasjdopj%20aspdojaspodjpoasjdpoawsjdpojawpdoj%20awsdwjpaowsjdpoawjdpoawsjdpoawjdopawd%20dpoasjdpoawsjdpoasjdpoasd&knob-The%20default%20content%20of%20the%20trigger%20button%20(trigger-content)=Select%20an%20item&knob-The%20notification%20kind%20(kind)=info&knob-Title%20(title)=Notification%20title&knob-Use%20low%20contrast%20variant%20(low-contrast)=true

### Steps to reproduce the issue (if applicable)

_No response_

### Release date (if applicable)

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

### PR DESCRIPTION
**Pull request by bruno-amorim**
### Related Ticket(s)

Closes #10730

https://jsw.ibm.com/browse/ADCMS-3670

### Description

When running the Accessibility checker on bx-breadcrumb and bx-breadcrumb-items, it complains about the bx-breadcrumb assigned role, where it is role="nav" saying that "The The role='nav' defined on the element is not valid per ARIA specification".
 
### Changelog

**New**

breadcrumb.ts code edited to now define the aria role as role="navigation"

### CODE REVIEW

**Comment by bruno-amorim** [comment]
I have read the DCO document and I hereby sign the DCO.

File: packages/carbon-web-components/src/components/breadcrumb/breadcrumb.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ const { prefix } = settings;
 class BXBreadcrumb extends LitElement {
   connectedCallback() {
     if (!this.hasAttribute('role')) {
-      this.setAttribute('role', 'nav');
+      this.setAttribute('role', 'navigation');
```
Comment (andy-blum) [inline packages/carbon-web-components/src/components/breadcrumb/breadcrumb.ts:26]: 👏

**Review by andy-blum — APPROVED** [review]
Thanks for your first contribution, @bruno-amorim!

**Review by annawen1 — APPROVED** [review]

**Comment by andy-blum** [comment]
Screenshot 2023-08-03 at 9 28 31 AM

Test failure appears to be unrelated. Closing/re-opening to make it try again.

**Review by RichKummer — APPROVED** [review]

**Pull request merged**